toqito/matrix_ops/tensor_comb.py

```diff
@@ -7,7 +7,12 @@
 from toqito.matrix_ops import to_density_matrix
 
 
-def tensor_comb(states: list[np.ndarray], k: int) -> dict:
+def tensor_comb(
+    states: list[np.ndarray],
+    k: int,
+    mode: str = "injective",
+    density_matrix: bool = True
+) -> dict[tuple[int, ...], np.ndarray]:
```

Comment:
For the examples, can you please use `.. jupyter-execute::` blocks (as is done in, for instance):
https://github.com/vprusso/toqito/blob/e55a29f01efe23c9447f80ee4f42de2e6bdf75f2/toqito/matrices/cyclic_permutation_matrix.py#L16

Reply:
Might need to rebase the PR for this.